pkg/activator/handler/context.go

```diff
@@ -33,21 +34,40 @@ type (
 	revIDKey    struct{}
 )
 
-// WithRevision attaches the Revision object to the context.
-func WithRevision(ctx context.Context, rev *v1.Revision) context.Context {
-	return context.WithValue(ctx, revisionKey{}, rev)
+type revCtx struct {
```

Comment:
out of interest, did you consider just doing `context.WithValue(ctx, key{}, struct{rev Revision, id string}{rev, revID})` and `ctx.Value(key{}).rev`/`ctx.Value(key{}).id` rather than a custom context type? Does the custom type have advantages over storing a struct as the value?

Reply:
Hmm, good question. Lemme try!

Reply:
Works the same, is a touch slower though, but simpler. Pushing.